Update several openvox-server files (#213)
OpenVoxProject/openvox-docs: docs/_openvox-server_8x/deprecated_features.markdown

---
layout: default
title: "OpenVox Server: Deprecated Features"
---

The following features / configuration settings are deprecated and will be removed in a future major release of OpenVox Server.

## `certificate-status` settings

### Now

If the `certificate-authority.certificate-status.authorization-required` setting is `false`, all requests that are successfully validated by SSL (if applicable for the port settings on the server) are permitted

> to use the [Certificate Status](https://github.com/openvoxproject/penvox/blob/master/api/docs/http_certificate_status.md) HTTP API endpoints. This includes requests which do not provide an SSL client certificate.

Reviewer: ```suggestion
to use the [Certificate Status](https://github.com/openvoxproject/openvox/blob/master/api/docs/http_certificate_status.md) HTTP API endpoints. This includes requests which do not provide an SSL client certificate.
```

### Now

If the `certificate-authority.certificate-status.authorization-required` setting is `false`, all requests that are successfully validated by SSL (if applicable for the port settings on the server) are permitted

> o use the [Certificate Status](https://github.com/openvoxproject/openvox/blob/master/api/docs/http_certificate_status.md) HTTP API endpoints. This includes requests which do not provide an SSL client certificate.

Reviewer: ```suggestion
to use the [Certificate Status](https://github.com/openvoxproject/openvox/blob/master/api/docs/http_certificate_status.md) HTTP API endpoints. This includes requests which do not provide an SSL client certificate.
```

The `environment_classes` endpoint ignores OpenVox's Ruby-based authorization methods and configuration in favor of OpenVox Server's Trapperkeeper authorization. For more information, see the
["Authorization" section](./puppet-api/v3/environment_classes.html) of the environment classes API documentation.

### Context

Users often rely on the `resource_types` endpoint for lists of classes and associated parameters in an environment. For such requests, the `resource_types` endpoint is inefficient and can trigger problematic
events, such as manifests being parsed during a catalog request.

To fulfill these requests more efficiently and safely, OpenVox Server 2.3.0 introduced the narrowly defined `environment_classes` endpoint.

## OpenVox's node cache terminus

### Now

OpenVox 5.0 (and by extension, OpenVox Server 5.0) no longer writes node YAML files to its cache by default.

### Previously

OpenVox wrote YAML to its node cache.

### Detecting and Updating

To retain the OpenVox 4.x behavior, add the [`puppet.conf`](./configuration.html) setting `node_cache_terminus = write_only_yaml`. The `write_only_yaml` option is deprecated.

### Context

> This cache was used in workflows where external tooling needs a list of nodes. OpenVox-DB is the preferred source of node information.

Reviewer: ```suggestion
This cache was used in workflows where external tooling needs a list of nodes. OpenVoxDB is the preferred source of node information.
```
